packages/source-iotsitewise/src/completeDataStreams.ts

```diff
@@ -13,34 +16,93 @@ const toDataType = (propertyDataType: PropertyDataType | string | undefined): Da
   return 'NUMBER';
 };
 
+const parseAlarmData = ({ alarmSource, data }: { alarmSource: AlarmSource; data: DataPoint[] }): DataPoint[] => {
+  if (alarmSource === IoTEventsSource) {
+    return data.map(({ x, y }: DataPoint): DataPoint => {
+      if (typeof y === 'string') {
+        return { x, y: parseIoTEventsAlarmData(y) };
+      }
+      return { x, y };
+    });
+  }
+
+  return data;
+};
+
 /**
  * Get completed data streams by merging together the data streams with the asset models.
  */
 export const completeDataStreams = ({
   dataStreams,
   assetModels,
+  alarms,
 }: {
   dataStreams: DataStream[];
   assetModels: Record<string, DescribeAssetModelResponse>;
-}): DataStream[] =>
-  dataStreams.map((dataStream) => {
+  alarms: AlarmsStore;
```

Comment:
is there a reason we need to pass around `AlarmStore` rather than `Alarm[]`?

Reply:
yeah we could definitely just do that and transform the AlarmStore to a list of Alarm in subscribeToDataStream

Reply:
Actually I think we might want to just stick with the store. We are doing the same for assetModels. Search is also less complex this way since I want to get the alarm by alarmStreamId in most cases (only need to convert to array for associatedStreams)